### PR TITLE
feat: add systemd unit file to enable apisix automatically

### PR DESCRIPTION
**Pull request by Yiyiyimu**
Signed-off-by: yiyiyimu <wosoyoung@gmail.com>

Test on CentOS7 and works
image

### CODE REVIEW

**Comment by Yiyiyimu** [comment]
@spacewander PTAL

File: usr/lib/systemd/system/apisix.service [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+Description=apisix
+Conflicts=apisix.service
+After=network.target
+
+[Service]
+Type=forking
+WorkingDirectory=/usr/bin
+ExecStart=/usr/bin/apisix start
+ExecStop=/usr/bin/apisix stop
+ExecReload=/usr/bin/apisix restart
```
Comment (spacewander) [inline usr/lib/systemd/system/apisix.service:11]: Should be `/usr/bin/apisix reload`.
You can refer the doc: https://www.freedesktop.org/software/systemd/man/systemd.service.html

File: usr/lib/systemd/system/apisix.service [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=apisix
+Conflicts=apisix.service
+After=network.target
```
Comment (spacewander) [inline usr/lib/systemd/system/apisix.service:4]: See https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/
Should be network-online.target

**Review by spacewander — CHANGES_REQUESTED** [review]

File: usr/lib/systemd/system/apisix.service (hunk 1 above)
Comment (Yiyiyimu) [reply usr/lib/systemd/system/apisix.service]: Thanks for the information!

File: usr/lib/systemd/system/apisix.service (hunk 2 above)
Comment (Yiyiyimu) [reply usr/lib/systemd/system/apisix.service]: Thank you! Fixed

File: usr/lib/systemd/system/apisix.service [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=apisix
+Conflicts=apisix.service
+After=network-online.target
+
+[Service]
+Type=forking
+WorkingDirectory=/usr/bin
```
Comment (spacewander) [inline usr/lib/systemd/system/apisix.service:8]: Maybe we can use the same PWD with the Nginx process?
```suggestion
WorkingDirectory=/usr/local/apisix
```

**Review by starsz — COMMENTED** [review]
`usr/lib/systemd/system/apisix.service`
The filepath in this project maybe not good ?

File: usr/lib/systemd/system/apisix.service (hunk 3 above)
Comment (Yiyiyimu) [reply usr/lib/systemd/system/apisix.service]: Thanks for suggestion!

**Comment by Yiyiyimu** [comment]
> `usr/lib/systemd/system/apisix.service`
> The filepath in this project maybe not good ?

Hi @starsz That's the method to install the service with fpm, since it does not support flag like `rpm-systemd` like for deb

Ref: https://github.com/jordansissel/fpm/issues/1163#issuecomment-492066328

> In the meantime, you can (ab)use the --config-files option to mark a systemd unit file as a config file, and include. The unit file can either be in a relative path mirroring it's root destination (./etc/systemd/system/file) or an absolute path (/etc/systemd/system/file).
The advantage of this method over the above methods, is that the file will be tracked by RPM and be removed when the package is removed - no scripts required.

**Review by spacewander — APPROVED** [review]

**Pull request merged**